### PR TITLE
Add AzStorageAccount and AzStorageAccountEvent

### PR DESCRIPTION
**Pull request by susam**
Author: @renusrijith

### CODE REVIEW

File: cloudmarker/events/azstorageaccountevent.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+"""Microsoft storage account secure transfer event.
```
Comment (mitprasoon) [inline cloudmarker/events/azstorageaccountevent.py]: The class name `AzStorageAccountEvent` does not capture the nature of this event plugin. A more apt name would be `AzStorageAccountSecureTransferEvent`.
There can be other events associated with an Azure storage account e.g. an event when default network rule is not set to deny

File: cloudmarker/events/azstorageaccountevent.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        """
+
+
+def _get_az_storage_account_secure_transfer_event(com, ext):
+    """Evaluate Azure storage account property for insecure transfer enabled.
+
+    Arguments:
+        com (dict): Azure storage account record `com` bucket
+        ext (dict): Azure storage account record `ext` bucket
+        raw (dict): Azure storage account record `raw` bucket
```
Comment (mitprasoon) [inline cloudmarker/events/azstorageaccountevent.py]: `raw` is not an argument to the method `_get_az_storage_account_secure_transfer_event`

File: cloudmarker/events/azstorageaccountevent.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    Returns:
+        dict: An event record representing storage accounts with  secure
```
Comment (mitprasoon) [inline cloudmarker/events/azstorageaccountevent.py]: There are two whitespaces between `with` and `secure`

File: cloudmarker/events/azstorageaccountevent.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        transfer enabled set to false
```
Comment (mitprasoon) [inline cloudmarker/events/azstorageaccountevent.py]: This must end with a period (.)

File: cloudmarker/events/azstorageaccountevent.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    event_record = {
+        # Preserve the  properties from the storage account
+        # record because they provide useful context to
+        # locate the storage account that led to the event.
+        'ext': util.merge_dicts(ext, {
+            'record_type': 'storage_account_event'
+        }),
+        'com': {
+            'cloud_type': com.get('cloud_type'),
+            'record_type': 'storage_account_event',
```
Comment (mitprasoon) [inline cloudmarker/events/azstorageaccountevent.py]: Record type `storage_account_event` does not capture the nature of this plugin.
`storage_account_secure_transfer_required_disabled_event` is a better fit.

File: cloudmarker/events/azstorageaccountevent.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            'reference': reference,
+            'description': description,
+            'recommendation': recommendation,
+        }
+    }
+    _log.info('Generating storage_account_event; %r',
```
Comment (mitprasoon) [inline cloudmarker/events/azstorageaccountevent.py]: Similarly, here the event type needs to be changed.

File: cloudmarker/events/azstorageaccountevent.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
+"""Microsoft storage account secure transfer event.
+
+This module defines the :class:`AzStorageAccountEvent` class that identifies
+a storage account with secure transfer enabled  not set to true .
```
Comment (mitprasoon) [inline cloudmarker/events/azstorageaccountevent.py]: Must be 
a storage account with secure transfer required  not set to true.

**Comment by susam** [comment]
Addressed the review comments. Will merge this to `master` now.

**Pull request merged**